test/run-drun/destabilization-crash/destabilization-crash.mo

```diff
@@ -23,7 +23,8 @@ actor a {
        c -= 1;
      };
      if (Prim.rts_heap_size() < mb * 1024 * 1024) {
-      Prim.debugPrint(debug_show({heap_MB = Prim.rts_heap_size()/1024/1024}));
+      let partitionSizeMB = 32;
```

Comment:
Why the change?

Reply:
Good point. This is actually a difference arising from the different object header lengths due to the forwarding pointer, and it is unrelated to partition size. I tried to improve these changes:
* Rename `partitionSizeMB` to `toleranceMB` and set a smaller value of 4MB for `destabilization-crash` and 2MB for similar code in `rts-stack-pages`
* I also moved the change down to the forwarding pointer PR https://github.com/dfinity/motoko/pull/3829. Since the incremental GC is only running on this PR, certain changes in the tests of the lower PR are only preparation work for the higher PR.